File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        unique = set()
        while l <= r:
            if nums[l] not in unique:
                unique.add(nums[l])
                l += 1
            else: 
                nums.pop(l)
                r -= 1
        return len(unique)

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 1) return 1;
        int ans = 1;
        int k = 1;
        int res = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (res != nums[i]) {
                k = 1;
                res = nums[i];
                ans++;
            } else {
                if (k < 2) {
                    k++;
                    ans++;
                } else {
                    nums[i] = 100001;
                }
            }
        }
        sort(nums.begin(), nums.end());
        return ans;
    }
};